addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/discovery/LGWebOSDiscovery.java

```diff
@@ -0,0 +1,221 @@
+        discoveryManager.setPairingLevel(DiscoveryManager.PairingLevel.ON);
+        discoveryManager.addListener(this);
+        super.activate(configProperties); // starts background discovery
+    }
+
+    @Override
+    protected void deactivate() {
+        super.deactivate(); // stops background discovery
+        discoveryManager.removeListener(this);
+        discoveryManager = null;
+        DiscoveryManager.destroy();
+        Util.uninit();
+    }
+
+    // @Override
+    @Override
+    protected void startScan() {
+        // no adhoc scanning. Discovery Service runs in background, but re-discover all known devices incase they were
+        // deleted from the inbox.
+        for (ConnectableDevice d : discoveryManager.getCompatibleDevices().values()) {
+            thingDiscovered(createDiscoveryResult(d));
+        }
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        discoveryManager.start();
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        discoveryManager.stop();
+    }
+
+    // DiscoveryManagerListener
+
+    @Override
+    public void onDeviceAdded(DiscoveryManager manager, ConnectableDevice device) {
+        thingDiscovered(createDiscoveryResult(device));
+    }
+
+    @Override
+    public void onDeviceUpdated(DiscoveryManager manager, ConnectableDevice device) {
+        logger.debug("Device updated: {}", device);
+        thingRemoved(createThingUID(device));
+        thingDiscovered(createDiscoveryResult(device));
+    }
+
+    @Override
+    public void onDeviceRemoved(DiscoveryManager manager, ConnectableDevice device) {
+        logger.debug("Device removed: {}", device);
+        thingRemoved(createThingUID(device));
+    }
+
+    @Override
+    public void onDiscoveryFailed(DiscoveryManager manager, ServiceCommandError error) {
+        logger.warn("Discovery Failed {}", error.getMessage());
+    }
+
+    // Helpers for DiscoveryManagerListener Impl
+    private DiscoveryResult createDiscoveryResult(ConnectableDevice device) {
+        ThingUID thingUID = createThingUID(device);
+        DiscoveryResult result = DiscoveryResultBuilder.create(thingUID)
+                .withProperty(PROPERTY_IP_ADDRESS, device.getIpAddress()).withLabel(device.getFriendlyName()).build();
+        return result;
+    }
+
+    private ThingUID createThingUID(ConnectableDevice device) {
+        return new ThingUID(THING_TYPE_WEBOSTV, device.getIpAddress().replace('.', '_'));
+    }
+
+    public DiscoveryManager getDiscoveryManager() {
+        return this.discoveryManager;
+    }
+
+    // Context for connect sdk
+    private final String DATA_DIR = new File("etc" + File.separator + "connect_sdk").getAbsolutePath();
+
+    @Override
+    public String getDataDir() {
+        return DATA_DIR;
+    }
+
+    @Override
+    public String getApplicationName() {
+        return "Openhab";
+    }
+
+    @Override
+    public String getPackageName() {
+        return "org.openhab";
```

Comment:
Can you use getClass().getPackage().getName() here? Just to avoid hard coded values.

Reply:
this would return org.openhab.binding.lgwebos.internal.discovery. In combination with ApplicationName this is used by LG TVs to identify the client application.
We don't want the actual package name. We need a constant namespace here. That's why I think org.openhab is better, as it would not change when we refactor LGWebOSDiscovery. 

But I'll add a comment to the code, why.